src/orion/testing/algo.py

```diff
@@ -416,84 +400,83 @@ def get_id(point, ignore_fidelity=False, exp_id=None):
+    @pytest.mark.parametrize("seed", [123])
+    def test_seed_rng(self, seed: int):
         """Test that the seeding gives reproducibile results."""
-        algo = self.create_algo()
+        algo = self.create_algo(seed=seed)
 
-        seed = numpy.random.randint(10000)
-        algo.seed_rng(seed)
-        spy = self.spy_phase(mocker, num, algo, attr)
-        trials = algo.suggest(1)
-        trials[0].id != algo.suggest(1)[0].id
+        trial_a = algo.suggest(1)[0]
+        trial_b = algo.suggest(1)[0]
+        assert trial_a != trial_b
 
-        new_algo = self.create_algo()
-        new_algo.seed_rng(seed)
-        self.force_observe(algo.n_observed, new_algo)
-        assert trials[0].id == new_algo.suggest(1)[0].id
-
-        self.assert_callbacks(spy, num + 1, new_algo)
+        new_algo = self.create_algo(seed=seed)
+        assert new_algo.n_observed == algo.n_observed
+        trial_c = new_algo.suggest(1)[0]
+        assert trial_c == trial_a
 
-    @phase
-    def test_seed_rng_init(self, mocker, num, attr):
+    def test_seed_rng_init(self):
         """Test that the seeding gives reproducibile results."""
         algo = self.create_algo(seed=1)
+        state = algo.state_dict
 
-        spy = self.spy_phase(mocker, num, algo, attr)
-        trials = algo.suggest(1)
-        algo.suggest(1)[0].id != trials[0].id
+        first_trial = algo.suggest(1)[0]
+        second_trial = algo.suggest(1)[0]
+        assert first_trial != second_trial
 
         new_algo = self.create_algo(seed=2)
-        self.force_observe(algo.n_observed, new_algo)
-        assert new_algo.suggest(1)[0].id != trials[0].id
+        new_algo_state = new_algo.state_dict
 
-        new_algo = self.create_algo(seed=1)
-        self.force_observe(algo.n_observed, new_algo)
-        assert new_algo.suggest(1)[0].id == trials[0].id
+        different_seed_trial = new_algo.suggest(1)[0]
+        if _are_equal(new_algo_state, state):
+            assert different_seed_trial == first_trial
+        else:
+            assert different_seed_trial != first_trial
 
-        self.assert_callbacks(spy, num + 1, new_algo)
+        new_algo = self.create_algo(seed=1)
+        same_seed_trial = new_algo.suggest(1)[0]
+        assert same_seed_trial == first_trial
 
-    @phase
-    def test_state_dict(self, mocker, num, attr):
+    @pytest.mark.parametrize("seed", [123, 456])
+    def test_state_dict(self, seed: int):
         """Verify that resetting state makes sampling deterministic"""
-        algo = self.create_algo()
+        algo = self.create_algo(seed=seed)
 
-        seed = numpy.random.randint(10000)
-        algo.seed_rng(seed)
-        spy = self.spy_phase(mocker, max(num, 1), algo, attr)
         state = algo.state_dict
         a = algo.suggest(1)[0]
 
+        # NOTE: This is not necessarily true for all algorithms. For instance, if the algo doesn't
+        # have any RNG (e.g. GridSearch), this will fail.
```

Comment:
But the algo should be reproducible anyhow, so the first suggested trial should be the same. Along the same lines, we assume the initial state to be the same, and to the be same for both after sampling the first trial.

Reply:
This doesn't just test the initial state anymore, it tests at each phase. This condition is correct, right?
I'll rephrase the note a bit.

Reply:
Rephrased the note in ecba557eb89cddf71eeb540e0d4af43f94e1841a

Reply:
Oh, I did not realize the changes to the test due to the modification of create_algo. So previously we would have `algo` updated up to the phase, but the new_algo would be at initial phase (num=0). I think it would be best to keep it this way, to make sure algorithms can have their state updated from different phases instead of always just one sample away. For this `create_algo` would need to take as argument whether it should be from scratch or at current phase.

Reply:
Just to make sure we are not reinserting a blind spot here, do you remember which test made you find the bug in hyperband when you were modifying the test-suite?

Reply:
> I think it would be best to keep it this way

You mean the way I have it here? I'm confused.

Reply:
> For this create_algo would need to take as argument whether it should be from scratch or at current phase.

Why do you need this, if the test is parametrized to run on all phases? Would you mind explaining again?

Reply:
> Just to make sure we are not reinserting a blind spot here, do you remember which test made you find the bug in hyperband when you were modifying the test-suite?

Which bug? Sorry I'm a bit slow this morning I guess

Reply:
Fixed in c88d490e382d116532dd4735c7723d3479ec7aea

I still don't quite recall what bug you're referring to. :(

Reply:
It's ok, I don't think you told me exactly what it was back then. Let's hope it is fixed properly. I'm it is. :)